modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -84,24 +88,42 @@
      * HashMap<String,List<Slot>>. Key in that hash map is queue name. value is List of Slot
      * objects.
      */
-    private IMap<String, HashmapStringTreeSetWrapper> overLappedSlotMap;
+    private final IMap<String, HashmapStringTreeSetWrapper> overLappedSlotMap;
 
     /**
      * Keeps slot deletion safe zones for each node, as they are informed by nodes via thrift
      * communication. Used for safe zone calculation for cluster by Slot Manager
      */
-    private Map<String, Long> nodeInformedSlotDeletionSafeZones;
+    private final Map<String, Long> nodeInformedSlotDeletionSafeZones;
 
     //safe zone calculator
-    private SlotDeleteSafeZoneCalc slotDeleteSafeZoneCalc;
+    private final SlotDeleteSafeZoneCalc slotDeleteSafeZoneCalc;
 
-    private static Log log = LogFactory.getLog(SlotManagerClusterMode.class);
+    /**
+     * Most recent node that left the cluster.
+     */
+    private String removedNode;
+
+    private static final Log log = LogFactory.getLog(SlotManagerClusterMode.class);
+
+    /**
+     * Denotes whether a slot recovery task is scheduled
+     */
+    private boolean slotRecoveryScheduled;
 
+    /**
+     * Queues that need to be recovered  While a slot recovery is scheduled a submit
+     * slot comes to the given queue that queue will be removed from the scheduled task.
+     */
+    private Set<String> queuesToRecover;
 
+    /**
+     * Creates a slot manager for cluster mode slot coordination
+     */
     private SlotManagerClusterMode() {
 
         HazelcastAgent hazelcastAgent = HazelcastAgent.getInstance();
-
+        removedNode = "";
```

Comment:
We need a node id of the removed node. Currently, we don't depend on the member removed method to select the coordinator. Initialising this to an empty string here in case, scheduled task starts before the member removed is called.
Anyway, this will get removed once we move the coordinator selection logic to member removed and added events